presto-main/src/main/java/com/facebook/presto/server/TaskInfoResource.java

```diff
@@ -69,4 +118,34 @@ public TaskInfo getTaskInfo(@PathParam("taskId") TaskId taskId)
             throw new NotFoundException(e);
         }
     }
+
+    private void proxyTaskInfo(HttpServletRequest servletRequest, AsyncResponse asyncResponse, String xForwardedProto, UriInfo uriInfo)
+    {
+        try {
+            checkState(proxyHelper.isPresent());
+            Optional<InternalNode> resourceManager = internalNodeManager.getResourceManagers().stream()
+                    .findFirst();
+            if (resourceManager.isPresent()) {
+                InternalNode resourceManagerNode = resourceManager.get();
+                URI uri = uriInfo.getRequestUriBuilder()
+                        .queryParam("includeLocalQueryOnly", true)
+                        .scheme(resourceManagerNode.getInternalUri().getScheme())
+                        .host(resourceManagerNode.getHostAndPort().toInetAddress().getHostName())
+                        .port(resourceManagerNode.getInternalUri().getPort())
+                        .build();
+                proxyHelper.get().performRequest(servletRequest, asyncResponse, uri);
+            }
+            else {
+                asyncResponse.resume(Response.status(SERVICE_UNAVAILABLE).build());
+            }
+        }
+        catch (Exception e) {
+            asyncResponse.resume(e);
```

Comment:
Can we return a meaningful error back to the client.?
Do you know what is the error response that is sent back in this case.

Reply:
We can return the error message with
```
asyncResponse.resume(Response.status(NOT_FOUND).entity("Could not find the requested taskInfo").build());
```
```
asyncResponse.resume(Response.serverError().entity(e.getMessage()).build());
```